### ISSUE TITLE
Accessibility: new Automation designer: unable to read choices for searches for trigers, conditons and actions

### ISSUE THREAD

**Issue by CityDweller** [home-assistant/frontend#19305]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

In the new automation designer in 2024.1, the drop down option to pick a triger, condition, or action has changed to a search box with results of that search.

The results are not readable/viewable with my screen reader (JAWS).

### Describe the behavior you expected

If we are going with the search approach, I would expect the same behavior as when you search/slect an entity/area...

You start typing and a filtered list is available by just down arrowing in the list.

### Steps to reproduce the issue

adding to @steverep's radar
- go to settings / automation
- select create new automation
- select add new trigger
- type 'time' inteh search box

The results for options that include time are not found by the screen reader.

### What version of Home Assistant Core has the issue?

2024.1.0

### What was the last working version of Home Assistant Core?

2023.12.4

### In which browser are you experiencing the issue with?

Microsoft Edge Version 120.0.2210.91

### Which operating system are you using to run this browser?

Microsoft Windows Version 23H2 (OS Build 22631.2861)

**Comment by CityDweller** [comment]
Found a workaround; 

after entering the search value, I can tab to the list box that contains choices if available

Not ideal, but acceptable to use

**Comment by Plawasan** [comment]
Not 100% sure this is the same issue but I'm missing the "state" input completely in the visual editor but the value is still there in YAML (two different conditions shown but they both behave the same)

![image](https://github.com/home-assistant/frontend/assets/73796046/7e476607-befc-4495-9ed0-707c29d2d6f2)

**Comment by steverep** [comment]
@CityDweller this isn't so much a bug as a deliberate change in the design.  

The old design was mechanically just a button menu (i.e. hit the button and arrow through the choices displayed).  Those were growing too big and disorganized with limited info for each choice.

The new design is a modal dialog with a search box at the top and a list box of options below it, similar to the dialog to add a new integration.  Focus does start in the search box, so yes you have to tab to the listbox of possibly filtered choices.

I do take your point well though, and agree the design is not ideal for screen reader users.  The roadmap in my head at least does include better a11y in general for these types of search and pick patterns.  For example, we have no feedback on search filter beyond what is happening visibly, which is a WCAG violation.  I'll try to come up with a concrete proposal to benefit many places in the UI and write it up.

**Comment by CityDweller** [comment]
Thanks for the response and clarification.
